lib/facter/facts/linux/az_metadata.rb

```diff
@@ -6,7 +6,7 @@ class AzMetadata
       FACT_NAME = 'az_metadata'
 
       def initialize
-        @virtual = Facter::Util::Facts::Posix::VirtualDetector.new
+        @virtual = Facter::Util::Facts::Posix::VirtualDetector
```

Comment:
You can also drop this and use `Facter::Util::Facts::Posix::VirtualDetector` directly in `azure_hypervisor?`